tests/customize.sh

```diff
@@ -0,0 +1,304 @@
+#!/bin/bash
+set -xeuo pipefail
+PS4='${LINENO}: '
+
+timeout=10m
+
+rootdir="$(realpath $(dirname $0)/..)"
+fixtures="${rootdir}/fixtures/customize"
+prog="${rootdir}/target/debug/coreos-installer"
+
+tmpd=$(mktemp -d)
+trap 'rm -rf "${tmpd}"' EXIT
+cd "${tmpd}"
+
+artifactdir=$1; shift
+ln -s ${artifactdir}/*-live-kernel-x86_64 src-kernel
+ln -s ${artifactdir}/*-live-initramfs.x86_64.img src-initrd
+ln -s ${artifactdir}/*-live-rootfs.x86_64.img src-rootfs
+ln -s ${artifactdir}/*-live.x86_64.iso src-iso
+
+digest() {
+    # Ignore filename
+    sha256sum "${1:--}" | awk '{print $1}'
+}
+
+# kargs we need on every boot
+kargs_common=(
+    # make sure we get log output
+    console=ttyS0,115200
+    # make sure it's readable
+    systemd.log_color=0
+    # force NM in initrd
+    rd.neednet=1
+)
+
+# options that don't initiate an install, even if they pertain to one
+opts_common=(
+    # @applied-live-ign@
+    # @did-not-install@ if that's true
+    --live-ignition "${fixtures}/live.ign"
+    # @applied-live-2-ign@
+    --live-ignition "${fixtures}/live-2.ign"
+    # @preinst-1@
+    --pre-install "${fixtures}/pre-install-1"
+    # @preinst-2@
+    --pre-install "${fixtures}/pre-install-2"
+    # @postinst-1@
+    --post-install "${fixtures}/post-install-1"
+    # @postinst-2@
+    --post-install "${fixtures}/post-install-2"
+    # 'Adding "coreos-installer test certificate" to list of CAs'
+    --ignition-ca "${fixtures}/cert.pem"
+    # Condition of @applied-live-ign@ and @applied-dest-ign@
+    --network-keyfile "${fixtures}/installer-test.nmconnection"
+)
+
+# options that do initiate an install
+opts_install=(
+    # @applied-dest-ign@
+    --dest-ignition "${fixtures}/dest.ign"
+    # @applied-dest-2-ign@
+    --dest-ignition "${fixtures}/dest-2.ign"
+    # Condition of @applied-dest-ign@
+    --installer-config "${fixtures}/install-1.conf"
+    # Condition of @applied-dest-ign@
+    --installer-config "${fixtures}/install-2.conf"
+    # Tested implicitly
+    --dest-device "/dev/vda"
+    # Condition of @applied-dest-ign@
+    --dest-karg-append dest-karg
+    # Condition of @applied-dest-ign@
+    --dest-karg-delete ignition.platform.id=metal
+    # Condition of @applied-dest-ign@
+    --dest-karg-append ignition.platform.id=qemu
+)
+for arg in ${kargs_common[@]}; do
+    opts_install+=(--dest-karg-append "${arg}")
+done
+
+opts_iso=()
+for arg in ${kargs_common[@]}; do
+    opts_iso+=(--live-karg-append "${arg}")
+done
+
+iso_customize() {
+    rm -f iso
+    "${prog}" iso customize src-iso -o iso "${opts_iso[@]}" "$@"
+}
+
+pxe_customize() {
+    rm -f initrd
+    "${prog}" pxe customize src-initrd -o initrd "$@"
+}
+
+qemu_common() {
```

Comment:
Would've been nice to leverage `kola qemuexec` for this, but yeah... it'd require a bunch of changes. Might be worth thinking about long-term doing the enhancements necessary and switching over. E.g. I'd love to be able to do e.g. `kola qemuexec -p pxe`.